sample_cgfs/rv32i_priv.cgf

```diff
@@ -1,21 +1,413 @@
+    (mstatus >> 36) & 0x1 == 0x01 : 0 			# check SBE = 1, it controls whether explicit load and store memory accesses made from S-mode is big-endian  
+    
+    
+
+mstatus_Mmode:    
+  config: 
+    - check ISA:=regex(.*I.*Zicsr.*); 
+  mnemonics: 
+    csrrw: 0
+    csrrs: 0
+  csr_comb:
+    (mstatus >> 3) & 0x01 == 0x01 : 0 			# checking if the MIE gets enabled
+    (mstatus >> 7) & 0x01 == 0x01 : 0 			# MPIE holds the value of the interrupt-enable bit active prior to the trap, checking whether it does that or not
+    (mstatus >> 10) & 0x3 == 0x3 : 0				# The xPP fields can only hold privilege modes up to x, so MPP is two bits wide, checking when if it becomes 11
+    (mstatus >> 10) & 0x2 == 0x2 : 0				# The xPP fields can only hold privilege modes up to x, so MPP is two bits wide, checking when if it becomes 10  
+    (mstatus >> 10) & 0x1 == 0x1 : 0				# The xPP fields can only hold privilege modes up to x, so MPP is two bits wide, checking when if it becomes 01  
+    (mstatus >> 17) & 0x1 == 0x01 : 0 			# checking MPRV, When MPRV=1, load and store memory addresses are translated and protected, and endianness is applied
+    (mstatus >> 19) & 0x1 == 0x01 : 0 			# check when MXR == 1    
+    (mstatus >> 37) & 0x1 == 0x01 : 0 			# check MBE = 1, it controls whether non-instruction-fetch memory accesses made from M-mode is big-endian
+
+
+mstatus_Umode:    
+  config: 
+    - check ISA:=regex(.*I.*Zicsr.*); 
+  mnemonics: 
+    csrrw: 0
+    csrrs: 0
+  csr_comb:
+    (mstatus >> 32) & 0x3 == 0x3 : 0				# checking UXL bits, If MXLEN is changed from 32 to a wider width, each of mstatus fields SXL and UXL, if not restricted to a single 									# value, gets the value corresponding to the widest supported width not wider than the new MXLEN.
+    (mstatus >> 18) & 0x1 == 0x01 : 0 			# check When SUM=1.  accesses to pages that are accessible by U-mode are permitted.
+    (mstatus >> 6) & 0x1 == 0x01 : 0 				# check UBE = 1, it controls whether explicit load and store memory accesses made from U-mode is big-endian
+
+
+mstatus:    
+  config: 
+    - check ISA:=regex(.*I.*Zicsr.*); 
+  mnemonics: 
+    csrrw: 0
+    csrrs: 0
+  csr_comb:
+    (mstatus >> 36) & (mstatus >> 37) & (mstatus >> 6) == 0 : 0 # when MBE = 0 and SBE = 0 and UBE = 0, accesses made are little endian
+    (mstatus >> 20) & 0x1 == 0x01 : 0 			# check when TVM = 1.
+    (mstatus >> 21) & 0x1 == 0x01 : 0 			# check when TW = 1.    
+
+    (mstatus & 0x3000) | (mstatus & 0x300) == (mstatus >> 31) : 0 # The SD bit is read-only and is set when either the FS, VS bits encode a Dirty state
+    
+mstatus_checking_SD:    
+  config: 
+    - check ISA:=regex(.*I.*Zicsr.*F.*V); 
+  mnemonics: 
+    csrrw: 0
+    csrrs: 0
+  csr_comb: 
+    (mstatus & 0x3000) | (mstatus & 0x300) == (mstatus >> 31) : 0 # The SD bit is read-only and is set when either the FS, VS bits encode a Dirty state 
```

Comment:
this looks like a property rather than a coverpoint. It would make sense to just check if SD was observed to 0 and 1

Reply:
resolved